woocommerce/woocommerce-gutenberg-products-block
# Storybook and TS migration of `ErrorPlaceholder` component

This PR adds stories for the `ErrorPlaceholder` component and migrates it to TypeScript.

Fixes #5255

### Testing

### Manual Testing

How to test the changes in this Pull Request:

1. Run `npm run storybook`
2. Explore the [Errors stories](http://localhost:6006/?path=/story/woocommerce-blocks-editor-components-errors-base-error-atom--base-error-atom). You should see five stories:

   * A base one with just the error message
   * The default placeholder with a generic error
   * Variation with an API type error
   * Variation with unknown error
   * Variation without retry button

In all the variations with the retry button, you should be able to click it and it will simulate a loading event for 3.5s.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
